scripts/generate_upgrade_syncset.py

```diff
@@ -0,0 +1,301 @@
+            'desired': {
+                'version': dq(version),
+                'channel': dq(channel),
+                'force': UC_FORCE,
+            }
+        }
+    }
+    return uc
+
+
+def generate_syncset(upgradeconfig, cluster_id, cluster_name='', environment='production'):
+    """
+    Generates the SelectorSyncSet
+    :param upgradeconfig: UpgradeConfig CR as dict
+    :param cluster_id: Cluster ID to target
+    :param cluster_name: Cluster name to target
+    :return: SelectorSyncSet as dict
+    """
+
+    # generate the namespace for the SyncSet in hive format
+    # uhc-<environment>-<clusterID>
+    ss_namespace = 'uhc-{}-{}'.format(environment, cluster_id)
+
+    data = CM()
+    data['apiVersion'] = SS_API_VERSION
+    data['kind'] = 'SyncSet'
+    data['metadata'] = {
+        'labels': {
+            'api.openshift.com/id': cluster_id,
+            'api.openshift.com/name': cluster_name,
+        },
+        'name': SS_NAME,
+        'namespace': ss_namespace,
+    }
+
+    data['spec'] = {
+        'clusterDeploymentRefs': [
+            {
+                'name': dq(cluster_name),
+            }
+        ],
+        'resourceApplyMode': 'Upsert',
+        'resources': [
+            upgradeconfig
+        ],
+    }
+
+    yaml.preserve_quotes = True
+    yaml.dump(data, sys.stdout, Dumper=yaml.RoundTripDumper)
+
+
+def init_logging():
+    logging.basicConfig(level=logging.INFO, format='%(asctime)s %(levelname)s:%(name)s:%(message)s')
+
+
+def generate_syncsets(clusters, start_time, version, channel, environment='production'):
+    outer = CM()
+    outer['apiVersion'] = 'v1'
+    outer['kind'] = 'List'
+
+    items = CS()
+    for cluster_id, cluster_name in clusters.items():
+        uc = generate_upgradeconfig(start_time, version, channel)
+        ss_namespace = 'uhc-{}-{}'.format(environment, cluster_id)
+        data = CM()
+        data['apiVersion'] = SS_API_VERSION
+        data['kind'] = 'SyncSet'
+        data['metadata'] = {
+            'labels': {
+                'api.openshift.com/id': cluster_id,
+                'api.openshift.com/name': cluster_name,
+            },
+            'name': SS_NAME,
+            'namespace': ss_namespace,
+        }
+
+        data['spec'] = {
+            'clusterDeploymentRefs': [
+                {
+                    'name': dq(cluster_name),
+                }
+            ],
+            'resourceApplyMode': 'Upsert',
+            'resources': [
+                uc
+            ],
+        }
+        items.append(data)
+
+    outer['items'] = items
+
+    yaml.preserve_quotes = True
+    yaml.dump(outer, sys.stdout, Dumper=yaml.RoundTripDumper)
+
+
+def init_argparse():
+    """
+    Initialises the command-line argument parser
+    :return: parser instance
+    """
+    parser = argparse.ArgumentParser(description='UpgradeConfig SelectorSyncSet generator',
+                                     usage='%(prog)s [options]')
```

Comment:
Why are you overriding `usage`? Without it, argparse generates a pretty useful usage statement:

```
usage: generate_upgrade_syncset.py [-h] [--subgroup {all,prod,nonprod}]
                                   [--start-time START_TIME] --version VERSION
                                   --channel CHANNEL [--force FORCE]
                                   [--environment {production,staging,integration}]
                                   [--cluster-id CLUSTER_ID]
                                   [--group {sd,sre,cicd,internal,snowflake,external}]
```

Reply:
Thanks, agree - I've just left it as-is to use the default in commit e1449f7